libchisel/src/utils.rs

```diff
@@ -0,0 +1,33 @@
+//! These are helpers to be used internally.
+
+use parity_wasm::elements::{deserialize_buffer, serialize, Module};
+
+pub trait HasNamesSection {
+    /// Returns true if the module has a NamesSection.
+    fn has_names_section(&self) -> bool;
+}
+
+impl HasNamesSection for Module {
+    fn has_names_section(&self) -> bool {
+        // Lets forcefully parse in case it wasn't yet.
+        let mut module = self.clone().parse_names().unwrap();
+        module.names_section().is_some()
+    }
+}
+
+pub trait SerializationHelpers {
+    /// Deserialize bytecode to a Module.
+    fn from_slice(input: &[u8]) -> Module;
+    /// Serialize Module to bytecode. Serialization consumes the input.
+    fn to_vec(self) -> Vec<u8>;
+}
+
+impl SerializationHelpers for Module {
+    fn from_slice(input: &[u8]) -> Self {
+        deserialize_buffer::<Module>(&input).expect("valid Wasm bytecode")
```

Comment:
On failure, should this not be `invalid Wasm bytecode`? same for below.

Reply:
Dunno, "expect valid wasm bytecode" sounded nice.

Reply:
I think it makes sense to do "Invalid Wasm bytecode/module" as the panic message does not say "Expected x". may cause confusion.